### PR TITLE
Add feature `kyber` for Round 3 Kyber API

### LINKED ISSUE #325: Offer Round 3 Kyber API in libcrux-ml-kem

### PR DESCRIPTION
**Pull request by jschneider-bensch**
This PR fixes #325, by making available `kyber{512,768,1024}`  modules with their own `encapsulate` and `decapsulate` behind the `kyber` feature, in addition to the regular ML-KEM API.
The differences in Kyber compared to ML-KEM are
- a hash of the initial encapsulation randomness is used instead of using the randomness directly
- the derivation of the shared secret includes a hash of the ciphertext

c.f. [FIPS 203 (Draft), section 1.3](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.203.ipd.pdf).

Further, the hybrid KEMs in `libcrux-kem` are now available in both variants as well.

### CODE REVIEW

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
I need to figure out how to make eurydice happy here. But I left some comments to be addressed.

**Review by franziskuskiefer — COMMENTED** [review]
With [this change](https://github.com/cryspen/libcrux/commit/8cdead9d4e2f022fbe794f5b3dd59c5df3ff0797) C extraction should work again. Looks like eurydice prefers the trait bounds on the functions. Let's do that for now.

File: libcrux-ml-kem/tests/kyber.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+use libcrux_ml_kem::mlkem768;
+
+#[test]
+#[cfg(feature = "kyber")]
+fn kyber768() {
```
[the remark this answers is not in the document]
Comment (jschneider-bensch) [reply libcrux-ml-kem/tests/kyber.rs]: I've added KAT generation for all parameter sets of Kyber, as we have it for ML-KEM as well.

**Review by franziskuskiefer — APPROVED** [review]
Thanks, lgtm. But we should also update the C code. And it looks like that extraction is broken again 😬

**Pull request merged**